transport-classes-kqueue/src/main/java/io/netty/channel/kqueue/KQueueEventLoop.java

```diff
@@ -353,6 +355,15 @@ public int registeredChannels() {
         return channels.size();
     }
 
+    @Override
+    public Iterator<Channel> registeredChannelsIterator() {
+        IntObjectMap<AbstractKQueueChannel> ch = channels;
+        if (ch.isEmpty()) {
+            return ChannelsReadOnlyIterator.empty();
+        }
+        return new ChannelsReadOnlyIterator<AbstractKQueueChannel>(ch.values());
```

Comment:
I think this is not thread-safe as calling values() is not thread-safe

Reply:
discussed above